File: components/prudence-foundation/libraries/scripturian/prudence/dispatch/python.py
#
# This file is part of the Prudence Foundation Library
#
# Copyright 2011-2012 Three Crickets LLC.
#
# The contents of this file are subject to the terms of the LGPL version 3.0:
# http://www.gnu.org/copyleft/lesser.html
#
# Alternatively, you can obtain a royalty free commercial license with less
# limitations, transferable or non-transferable, directly from Three Crickets
# at http://threecrickets.com/
#

resources = {}

document.executeOnce(application.globals['prudence.dispatch.python.library'])

def handle(conversation, method):
    id = conversation.locals['prudence.id']
    resource = resources.get(id, None)
    if resource is None:
        conversation.statusCode = 404
        return None
    method = getattr(resource, method, None)
    if method is None:
        conversation.statusCode = 405
        return None
    return method(conversation)

def handle_init(conversation):
    handle(conversation, 'handle_init')

def handle_get(conversation):
    return handle(conversation, 'handle_get')

def handle_get_info(conversation):
    return handle(conversation, 'handle_get_info')

def handle_post(conversation):
    return handle(conversation, 'handle_post')

def handle_put(conversation):
    return handle(conversation, 'handle_put')

def handle_delete(conversation):
    return handle(conversation, 'handle_delete')

File: components/prudence/source/com/threecrickets/prudence/test/Python.java
/**
 * Copyright 2009-2012 Three Crickets LLC.
 * <p>
 * The contents of this file are subject to the terms of the LGPL version 3.0:
 * http://www.gnu.org/copyleft/lesser.html
 * <p>
 * Alternatively, you can obtain a royalty free commercial license with less
 * limitations, transferable or non-transferable, directly from Three Crickets
 * at http://threecrickets.com/
 */

package com.threecrickets.prudence.test;

import com.threecrickets.prudence.test.internal.DistributionTest;
import com.threecrickets.prudence.test.internal.TestOK;
import com.threecrickets.prudence.test.internal.TestRedirected;

/**
 * @author Tal Liron
 */
public class Python extends DistributionTest
{
	//
	// Construction
	//

	public Python()
	{
		super( "python", true );
	}

	// //////////////////////////////////////////////////////////////////////////
	// Protected

	@Override
	protected Runnable[] getPrudenceTests()
	{
		return prudenceTests;
	}

	// //////////////////////////////////////////////////////////////////////////
	// Private

	private static Runnable[] prudenceTests = new Runnable[]
	{
		new TestOK( "/prudence-test/" ), new TestRedirected( "/prudence-test" ),
		// /web/static/
		new TestOK( "/prudence-test/style/soft-cricket.css" ),
		// /web/dynamic/
		new TestOK( "/prudence-test/test/jython/?id=hello" ), new TestOK( "/prudence-test/test/velocity/?id=hello" ),
		// /resources/
		new TestOK( "/prudence-test/data/jython/" )
	};
}
